src/likelihoods/categorical.jl

```diff
@@ -1,22 +1,46 @@
 """
-    CategoricalLikelihood(l=softmax)
+    CategoricalLikelihood(l=softmax; bijective::Union{Bool,Val{false},Val{true}}=Val(true))
 
 Categorical likelihood is to be used if we assume that the 
-uncertainity associated with the data follows a Categorical distribution.
+uncertainty associated with the data follows a [Categorical distribution](https://en.wikipedia.org/wiki/Categorical_distribution) with `n` categories.
+
+Two variants are possible (you can use a `Bool` or `Val{true}`/`Val{false}` but
+we recommend the latter for type stability).
+## `bijective=true/Val{true}`
 ```math
     p(y|f_1, f_2, \\dots, f_{n-1}) = \\operatorname{Categorical}(y | l(f_1, f_2, \\dots, f_{n-1}, 0))
 ```
 Given an `AbstractVector` ``[f_1, f_2, ..., f_{n-1}]``, returns a `Categorical` distribution,
 with probabilities given by ``l(f_1, f_2, ..., f_{n-1}, 0)``.
+Under the hood, the link `l` is wrapped into a [`SimplexBijectiveLink`](@ref) which takes
+care of concatenating `0`.
+It is used by default.
+
+## `bijective=false`
+```math
+    p(y|f_1, f_2, \\dots, f_n) = \\operatorname{Categorical}(y | l(f_1, f_2, \\dots, f_n))
+```
+Given an `AbstractVector` ``[f_1, f_2, ..., f_{n}]``, returns a `Categorical` distribution,
+with probabilities given by ``l(f_1, f_2, ..., f_{n})``.
+This variant is over-parametrized, as there are `n-1` independent parameters 
+embedded in a `n` dimensional parameter space.
+For more details, see the end of the section of this [Wikipedia link](https://en.wikipedia.org/wiki/Exponential_family#Table_of_distributions)
+where it corresponds to Variant 1 and 2.
 """
 struct CategoricalLikelihood{Tl<:AbstractLink} <: AbstractLikelihood
     invlink::Tl
 end
 
-CategoricalLikelihood(l=softmax) = CategoricalLikelihood(link(l))
+function CategoricalLikelihood(
+    l=softmax; bijective::Union{Bool,Val{true},Val{false}}=Val(true)
+)
+    return CategoricalLikelihood(make_bijective(link(l), bijective))
+end
```

Comment:
What I meant: I don't think we should not introduce `bijective` at all - since it makes it more difficult to use a custom user-defined link function. Even though you code everything yourself you would have to deal with `bijective` and end up with a wrapper that's not needed.

More concretely, I thought something like this should be sufficient:
```julia
CategoricalLikelihood(l=BijectiveSimplexLink(softmax)) = CategoricalLikelihood(link(l))

(l::CategoricalLikelihood)(f::AbstractVector{<:Real}) = Categorical(l.invlink(f))

function (l::CategoricalLikelihood)(fs::AbstractVector)
    return Product(Categorical.(l.invlink.(fs)))
end

struct BijectiveSimplexLink{T}
    link::T
end

(l::BijectiveSimplexLink)(f::AbstractVector{<:Real}) = l.link(vcat(f, 0))
```
```